plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/internal/core/WatchManager.java

```diff
@@ -271,9 +331,9 @@ public void connectionRemoved(IConnection connection) {
 			Connection conn = (Connection)connection;
 			synchronized (watches) {
 				watches.keySet().stream()
-					.filter(p->conn.ownsResource(p))
+					.filter(k->k.connection.equals(conn))
```

Comment:
@fbricon @jcantrill @jeffmaury this was corrected quite some time ago. Connection#equals does NOT consider authentication info, it just compares username and server-urlhttps://github.com/adietish/jbosstools-openshift/blob/master/plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/connection/Connection.java#L626 You can even see this in the tests: https://github.com/adietish/jbosstools-openshift/blob/master/tests/org.jboss.tools.openshift.test/src/org/jboss/tools/openshift/test/core/connection/ConnectionTest.java#L203